Tests/SentryTests/SentrySessionTests.m

```diff
@@ -0,0 +1,128 @@
+#import <XCTest/XCTest.h>
+#import "SentrySession.h"
+
+@interface SentrySessionTests : XCTestCase
+
+@end
+
+@implementation SentrySessionTests
+
+- (void)testInitDefaultValues {
+    SentrySession *session = [[SentrySession alloc] init];
+    XCTAssertNotNil(session.sessionId);
+    XCTAssertEqual(1, session.sequence);
+    XCTAssertEqual(0, session.errors);
+    XCTAssertTrue(session.init);
+    XCTAssertNotNil(session.started);
+    XCTAssertEqual(kSentrySessionStatusOk, session.status);
+    XCTAssertNotNil(session.distinctId);
+
+    XCTAssertNil(session.timestamp);
+    XCTAssertNil(session.releaseName);
+    XCTAssertNil(session.environment);
+    XCTAssertNil(session.duration);
+}
+
+- (void)testSerializeDefaultValues {
+    SentrySession *expected = [[SentrySession alloc] init];
+    NSDictionary<NSString *, id>  *json = [expected serialize];
+    SentrySession *actual = [[SentrySession alloc] initWithJSONObject:json];
+
+    XCTAssertTrue([expected.sessionId isEqual:actual.sessionId]);
+    XCTAssertEqual(expected.sequence, actual.sequence);
+    XCTAssertEqual(expected.errors, actual.errors);
+    // TODO: get XCT happy XCTAssertEqual(expected.init, actual.init);
+    XCTAssertEqualWithAccuracy([expected.started timeIntervalSinceReferenceDate], [actual.started timeIntervalSinceReferenceDate], 1);
+    XCTAssertEqual(expected.status, actual.status);
+    XCTAssertEqual(expected.distinctId, actual.distinctId);
+    XCTAssertNil(expected.timestamp);
```

Comment:
I think we could drop this statement here, as we already check in `testInitDefaultValues` if the default value of `timestamp` is `nil`.

Reply:
I thought it wouldn't hurt because if the default changes later, it should point out the relationship between the default and initWithJSONObject but happy to remove this assert